### PR TITLE
[MI-2259] Created the modal for updating state

### PR DESCRIPTION
**Pull request by manojmalik20**
Created websocket events for triggering the opening of Update state modal
Created and registered websocket event handler for opening the update state modal
Created reducer for update state modal
Created the component for Update state modal and used dummy data for the dropdown

#### Screenshots
image

image

image

### CODE REVIEW

File: webapp/src/containers/updateState/index.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import React, {useCallback, useState} from 'react';
+import {useDispatch} from 'react-redux';
+import {CustomModal as Modal, Dropdown, ModalFooter, ModalHeader, ModalSubtitleAndError} from '@brightscout/mattermost-ui-library';
```
Comment (ayusht2810) [inline webapp/src/containers/updateState/index.tsx:3]:
```suggestion
import {useDispatch} from 'react-redux';

import {CustomModal as Modal, Dropdown, ModalFooter, ModalHeader, ModalSubtitleAndError} from '@brightscout/mattermost-ui-library';
```

File: webapp/src/containers/updateState/index.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import usePluginApi from 'hooks/usePluginApi';
+
+import {hideModal as hideUpdateStateModal} from 'reducers/updateStateModal';
+
+const stateOptions: DropdownOptionType[] = [
```
Comment (ayusht2810) [inline webapp/src/containers/updateState/index.tsx:9]: we can add a TODO comment here stating these are not integrated yet

File: webapp/src/containers/updateState/index.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const hideModal = useCallback(() => {
+        setSelectedState(null);
+        dispatch(hideUpdateStateModal());
+    }, []);
+
+    return (
+        <Modal
+            show={pluginState.openUpdateStateModalReducer.open}
+            onHide={hideModal}
+            className={'rhs-modal'}
```
Comment (ayusht2810) [inline webapp/src/containers/updateState/index.tsx:56]:
```suggestion
            className='rhs-modal'
```

File: webapp/src/containers/updateState/index.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        >
+            <>
+                <ModalHeader
+                    title={'Update State'}
```
Comment (ayusht2810) [inline webapp/src/containers/updateState/index.tsx:60]: we can remove {} from the strings. check everywhere

File: webapp/src/containers/updateState/index.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                />
+                <div className={'padding-h-12 padding-v-20 wizard__body-container'}>
+                    <Dropdown
+                        placeholder={'Select State'}
+                        value={selectedState}
+                        onChange={setSelectedState}
+                        options={stateOptions}
+                        required={true}
+                    />
+                    <ModalSubtitleAndError error={''}/>
```
Comment (ayusht2810) [inline webapp/src/containers/updateState/index.tsx:72]: if the error is always `''` we can remove this, right?

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/containers/updateState/index.tsx (hunk 2 above)
Comment (manojmalik20) [reply webapp/src/containers/updateState/index.tsx:9]: I don't really think it's necessary as it is already integrated in the next PR.

File: webapp/src/websocket/index.ts [hunk 6]
```diff
@@ -68,6 +69,18 @@ export function handleOpenShareRecordModal(store: Store<GlobalState, Action<Reco
     };
 }
 
+export function handleOpenCommentModal(store: Store<GlobalState, Action<Record<string, unknown>>>) {
+    return (msg: WebsocketEventParams) => {
+        // TODO: Fix the type of state below by importing the GlobalState from mattermost-webapp
```
Comment (ayusht2810) [inline webapp/src/websocket/index.ts:74]: do we need this comment here, as we have defined the type for msg already?

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/websocket/index.ts (hunk 6 above)
Comment (manojmalik20) [reply webapp/src/websocket/index.ts:74]: This comment is not for the type of `msg`, it is for the type of `GlobalState`.

File: webapp/src/containers/updateState/index.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+import {CustomModal as Modal, Dropdown, ModalFooter, ModalHeader} from '@brightscout/mattermost-ui-library';
+
+import usePluginApi from 'hooks/usePluginApi';
+
+import {hideModal as hideUpdateStateModal} from 'reducers/updateStateModal';
+
+const stateOptions: DropdownOptionType[] = [
+    {
+        label: 'New',
+        value: '1',
```
Comment (avas27JTG) [inline webapp/src/containers/updateState/index.tsx:13]: can create enums for numberings to have more readable code

File: webapp/src/reducers/updateStateModal/index.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+export const openUpdateStateModalSlice = createSlice({
+    name: 'openUpdateStateModalSlice',
+    initialState,
+    reducers: {
+        showModal: (state) => {
+            state.open = true;
+        },
+        hideModal: (state) => {
+            state.open = false;
+        },
```
Comment (avas27JTG) [inline webapp/src/reducers/updateStateModal/index.ts:20]:
```suggestion
        toggleModalVisibility: (state, action) => {
            state.open = action.payload;
        },
```

can pass `true/false` in action

File: webapp/src/containers/updateState/index.tsx (hunk 7 above)
Comment (manojmalik20) [reply webapp/src/containers/updateState/index.tsx:13]: This is just demo data and it is removed in the next PR.

File: webapp/src/reducers/updateStateModal/index.ts (hunk 8 above)
Comment (manojmalik20) [reply webapp/src/reducers/updateStateModal/index.ts:20]: Adding a TODO comment.

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**